test/controllers/close_contacts_controller_test.rb

```diff
@@ -97,6 +97,147 @@ def teardown; end
     sign_out user
   end
 
+  # --- INDEX --- #
+
+  test 'index: returns error if entries params cannot be validated' do
+    user = create(:public_health_enroller_user)
+    patient = create(:patient, creator: user)
+    create(:close_contact, patient: patient)
+
+    sign_in user
+
+    params = {
+      patient_id: patient.id,
+      entries: -1, # invalid param
+      page: 0,
+      search: '',
+      order: nil,
+      direction: nil
+    }
+
+    get :index, params: params
+
+    assert_response(:bad_request)
+    assert_equal("Invalid Query (entries): #{params[:entries]}", JSON.parse(response.body)['error'])
+
+    sign_out user
+  end
+
+  test 'index: returns error if page params cannot be validated' do
+    user = create(:public_health_enroller_user)
+    patient = create(:patient, creator: user)
+    create(:close_contact, patient: patient)
+
+    sign_in user
+
+    params = {
+      patient_id: patient.id,
+      entries: 5,
+      page: -1, # invalid param
+      search: '',
+      order: nil,
+      direction: nil
+    }
+
+    get :index, params: params
+
+    assert_response(:bad_request)
+    assert_equal("Invalid Query (page): #{params[:page]}", JSON.parse(response.body)['error'])
+
+    sign_out user
+  end
+
+  test 'index: returns bad_request if patient cannot be found' do
+    user = create(:public_health_enroller_user)
+    sign_in user
+
+    mock_invalid_id = 'test'
+
+    get :index, params: {
+      patient_id: mock_invalid_id,
+      entries: 10,
+      page: 0,
+      search: '',
+      order: nil,
+      direction: nil
+    }
+
+    assert_response(:bad_request)
+    assert_equal("Unknown patient with ID: #{mock_invalid_id&.to_i}", JSON.parse(response.body)['error'])
+
+    sign_out user
+  end
+
+  test 'index: redirects if patient has no close contacts' do
+    user = create(:public_health_enroller_user)
+    patient = create(:patient, creator: user) # no close contacts
+    sign_in user
+
+    get :index, params: {
+      patient_id: patient.id,
+      entries: 10,
+      page: 0,
+      search: '',
+      order: nil,
+      direction: nil
+    }
+    assert_equal([], JSON.parse(response.body)['table_data'])
+    assert_equal(0, JSON.parse(response.body)['total'])
+    # assert_equal("Unknown patient with ID #{mock_invalid_id&.to_i}", JSON.parse(response.body)['error'])
+    # assert_redirected_to(@controller.root_url)
```

Comment:
Are lines 186 & 187 suppose to be commented out?

Reply:
All these tests were modeled off the vaccine controller tests, which I didn't write, but they are commented out there too (see [here](https://github.com/SaraAlert/SaraAlert/blob/1527-labs-table-refactor/test/controllers/vaccines_controller_test.rb#L193)).  Not sure why they were left in but I assume it was important enough to I kept it in the new ones too